apps/platform-app/src/app/[locale]/(profile)/(components)/CVuploader/CvUploaderForm.tsx

```diff
@@ -4,25 +4,58 @@ import { useUploadContext } from '@/contexts/UploadContext'
 import { I18nNamespaces } from '@/i18n/request'
 import { Button } from '@gdh/ui-system'
 import { ErrorIcon } from '@gdh/ui-system/icons'
+import { useSession } from 'next-auth/react'
 import { useTranslations } from 'next-intl'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
+import { fetchCVurl } from '../../_actions/mutations/fetchCVurl'
 import styles from './CvUploaderForm.module.scss'
 
 export function CVuploaderForm() {
   const t = useTranslations(I18nNamespaces.Buttons)
+  const tt = useTranslations(I18nNamespaces.PersonalInfo)
+  const { data: session } = useSession()
   const [isUploading, setIsUploading] = useState(false)
+  const [uploadCvButtonText, setUploadCvButtonText] = useState<string>(
+    t('uploadCVfile'),
+  )
+
+  const [uploadedCVurl, setUploadedCVurl] = useState<{
+    name: string
+    url: string
+  } | null>(null)
 
   const { cvUploadError, onSetCvUploadError, onSetCvFormData } =
     useUploadContext()
-  const [selectedFile, setSelectedFile] = useState<File | null>(null)
+
   const [errorMsg, setErrorMsg] = useState<string | null>(null)
 
+  useEffect(() => {
```

Comment:
**nit:** Do we really need to fetch this URL here if its a part of ProfileModel which is fetched to form state? I think we can extract this url from form state